small method code cleanup

lcnr opened the pull request.

lcnr committed:
small method code cleanup

lcnr committed:
remove `TypeError::ObjectUnsafeCoercion`

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/3549074686/jobs/5961019879) [(plain)](https://github.com/rust-lang/rust/commit/6f2b9919fe4ae1ceef49b2d19b74d636b3d7b580/checks/9710179447/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    Checking rustc_hir_typeck v0.1.0 (/checkout/compiler/rustc_hir_typeck)
error: associated function `report_object_unsafe_cast` is never used
   --> compiler/rustc_hir_typeck/src/cast.rs:742:8
    |
742 |     fn report_object_unsafe_cast(&self, fcx: &FnCtxt<'a, 'tcx>, did: DefId) {
    |
    |
    = note: `-D dead-code` implied by `-D warnings`
error: could not compile `rustc_hir_typeck` due to previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `rustc_hir_typeck` due to previous error
Build completed unsuccessfully in 0:03:25
```

</details>

lcnr committed:
remove confusing comment

`?0: CoerceUnsized<SomeTy>` can definitely apply because
`?0` matches any type, same for `SomeTy: CoerceUnsized<?0>`

fee1-dead commented on compiler/rustc_hir_typeck/src/callee.rs:
```diff
@@ -179,12 +179,16 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
-            // `FnMut::call_mut(&mut *x, ())`.
+            // like `f: &mut FnMut()`, if there is a call `f()`, we would
+            // normally translate to `FnMut::call_mut(&mut f, ())`, but
+            // that winds up requiring the user to potentially mark their
```
```suggestion
            // that winds up potentially requiring the user to mark their
```

lcnr committed:
fix type

Co-authored-by: fee1-dead <ent3rm4n@gmail.com>

fee1-dead commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 99c3dda06649ae9878138366bf2423ce45ce198b has been approved by `fee1-dead`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=fee1-dead 99c3dda06649ae9878138366bf2423ce45ce198b -->
<!-- homu: {"type":"Approved","sha":"99c3dda06649ae9878138366bf2423ce45ce198b","approver":"fee1-dead","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
